GovUk.Frontend.AspNetCore.Extensions/Models/PaginationModel.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Microsoft.AspNetCore.Http;
+using System;
+
+namespace GovUk.Frontend.AspNetCore.Extensions.Models
+{
+    /// <summary>
+    /// Configuration settings for an instance of the GOV.UK Design System Pagination component.
+    /// </summary>
+    public class PaginationModel
+    {
+        /// <summary>
+        /// The current page.
+        /// </summary>
+        public int PageNumber { get; set; } = 1;
+
+        /// <summary>
+        /// The number of items per page. If <c>null</c> then pagination is not active.
+        /// </summary>
+        public int? PageSize { get; set; }
+
+        /// <summary>
+        /// The total number of items to be paginated.
+        /// </summary>
+        public int TotalItems { get; set; }
+
+        /// <summary>
+        /// The total number of pages based on <see cref="TotalItems"/> and <see cref="PageSize"/>
+        /// </summary>
+        public int TotalPages()
+        {
+            if (!PageSize.HasValue || PageSize <= 0 || TotalItems <= 0)
+            {
+                return 1;
+            }
+            return (int)Math.Ceiling((decimal)TotalItems / PageSize.Value);
```

Comment:
Would an explanation as to why we are using `PageSize.Value` rather than `PageSize` in the division help future users?